staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -329,7 +329,9 @@ func (o *WaitOptions) RunWait() error {
 		// or functions from ResourceBuilder for parsing those. Lastly, this poll
 		// should be replaced with a ListWatch cache.
 		if err := wait.PollUntilContextTimeout(ctx, 500*time.Millisecond, o.Timeout, true, func(context.Context) (done bool, err error) {
+			foundResource := false
```

Comment:
Thank you. That looks reasonable. But we definitely need unit (and/or integration) test to verify the behavior.

Reply:
Yes of course. Ill add a test to verify

Reply:
Add in https://github.com/kubernetes/kubernetes/pull/128662/commits/eaaf91f9af89860708122660e9d72783606e6ef2